include/nektar_interface/geometry_transport_3d.hpp

```diff
@@ -0,0 +1,586 @@
+  deconstruct_geoms_base_3d(rank, geometry_id, geom, deconstructed_geoms);
+  const int num_faces = geom->GetNumFaces();
+  for (int facex = 0; facex < num_faces; facex++) {
+    auto geom_2d = geom->GetFace(facex);
+    const int geom_2d_gid = geom_2d->GetGlobalID();
+    face_ids.insert(geom_2d_gid);
+  }
+}
+
+/**
+ *  Deconstruct, i.e. serialise to integers, 3D Nektar++ geometry objects such
+ *  that they can be sent over MPI. Geometry objects are packed into an output
+ *  map for each remote MPI rank.
+ *
+ *  @param[in] original_rank MPI rank which owns the geometry objects in
+ * `rank_element_map`.
+ *  @param[in] geoms_2d Map from geometry id to 2D geometry objects required by
+ *  the 3D geometry objects which are t o be packed.
+ *  @param[in] rank_element_map Map from remote MPI rank to map from geometry id
+ * to 3D geometry object to pack for the remote rank.
+ *  @param[in] send_ranks Vector of remote MPI ranks.
+ *  @param[in, out] send_sizes Output vector of packed sizes per MPI rank.
+ *  @param[in, out] deconstructed_geoms Map from remote MPI rank to vector of
+ *  ints describing the packed 3D geoms in terms of the 2D geometry ids that
+ *  build the faces of the 3D object.
+ *  @param[in, out] rank_triangle_map Map from remote MPI rank to TriGeom ids
+ * and objects to pack and send.
+ *  @param[in, out] rank_quad Map from remote MPI rank to QuadGeom ids and
+ * objects to pack and send.
+ */
+inline void deconstuct_per_rank_geoms_3d(
+    const int original_rank,
+    std::map<int, std::shared_ptr<Nektar::SpatialDomains::Geometry2D>>
+        &geoms_2d,
+    std::map<int,
+             std::map<int, std::shared_ptr<Nektar::SpatialDomains::Geometry3D>>>
+        &rank_element_map,
+    const int num_send_ranks, std::vector<int> &send_ranks,
+    std::vector<int> &send_sizes,
+    std::map<int, std::vector<int>> &deconstructed_geoms,
+    std::map<int,
+             std::map<int, std::shared_ptr<Nektar::SpatialDomains::TriGeom>>>
+        &rank_triangle_map,
+    std::map<int,
+             std::map<int, std::shared_ptr<Nektar::SpatialDomains::QuadGeom>>>
+        &rank_quad_map) {
+
+  // the face ids to be sent to each remote rank
+  std::map<int, std::set<int>> face_ids;
+  int rank_index = 0;
+  for (int rank : send_ranks) {
+    deconstructed_geoms[rank].reserve(7 * rank_element_map[rank].size());
+    for (auto &geom_pair : rank_element_map[rank]) {
+      deconstruct_geoms_3d(original_rank, geom_pair.first, geom_pair.second,
+                           deconstructed_geoms[rank], face_ids[rank]);
+    }
+    send_sizes[rank_index++] = deconstructed_geoms[rank].size();
+  }
+
+  //  We have now collected the set of 2D geoms to send to each remote rank
+  //  that are required to recreate the 3D geoms we will send to each rank.
+  //  The deconstructed geoms array describes how to recreate the 3D geoms.
+
+  // The 2D exchange routines exchange triangles and quads seperately so we
```

Comment:
```suggestion
  // The 2D exchange routines exchange triangles and quads separately so we
```
Suggest find-replace seperate=>separate